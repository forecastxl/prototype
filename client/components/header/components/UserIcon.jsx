import React, { PropTypes } from 'react'
import SocialPerson from 'material-ui/svg-icons/social/person'
import muiThemeable from 'material-ui/styles/muiThemeable'

export function UserIcon({ muiTheme }) {
  return (
    <SocialPerson style={{ color: muiTheme.palette.alternateTextColor }} />
  )
}

UserIcon.propTypes = {
  muiTheme: PropTypes.object.isRequired
}

export default muiThemeable()(UserIcon)
