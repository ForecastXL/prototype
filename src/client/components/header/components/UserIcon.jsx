import React from 'react'
import { object } from 'prop-types'
import SocialPerson from 'material-ui/svg-icons/social/person'
import muiThemeable from 'material-ui/styles/muiThemeable'

export function DumbUserIcon({ muiTheme }) {
  return <SocialPerson style={{ color: muiTheme.palette.alternateTextColor }} />
}

DumbUserIcon.propTypes = {
  muiTheme: object.isRequired
}

export default muiThemeable()(DumbUserIcon)
