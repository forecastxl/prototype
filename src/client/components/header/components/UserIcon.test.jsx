import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { DumbUserIcon } from './UserIcon'

describe('<DumbUserIcon />', () => {
  it('renders correctly', () => {
    const theme = {
      palette: {
        alternateTextColor: 'blue'
      }
    }
    const wrapper = shallow(<DumbUserIcon muiTheme={theme} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
