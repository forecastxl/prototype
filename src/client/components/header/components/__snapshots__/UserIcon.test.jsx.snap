// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DumbUserIcon /> renders correctly 1`] = `
<SocialPerson
  style={
    Object {
      "color": "blue",
    }
  }
/>
`;

exports[`<UserIcon /> renders correctly 1`] = `
<SocialPerson
  style={
    Object {
      "color": "blue",
    }
  }
/>
`;
